src/App.tsx

```diff
@@ -1,9 +1,58 @@
+import { useEffect } from "react";
 import styles from "./App.module.css";
+import {faker} from "@faker-js/faker";
 
-const App = () => (
+interface Flight {
+  flightNumber: string;
+  origin: string;
+  destination: string;
+  departureTime: Date;
+  arrivalTime: Date;
+  flightDuration: number;
+}
+
+
+
+const App = () => {
+
+  const [allFlights, setFlights] = useState(0);
+  
+
+  useEffect(() => {
+
+  })
```

Comment:
No worries that you didn't finish, but I'd like to show you how this would be done!

We can use the useEffect to "react" (get it!) to changes in whatever we put at the array at the end of the function. This is called a "dependency array" and it call run the callback function whenever a change is recognized in any one of the values in that array. 

In our case, we want sort the flights array any time we add one! So we can say that we want this callback function to run whenever the flight array changes and this callback with sort the array and replace it in the state.

```ts
useEffect(() => {

   // sort the flights array which now has a new value 
   const newFlights = allFlights.sort((a: Flight, b: Flight) => {
     return a.flightDuration - b.flightDuration
   }
   
   // replace the old flights array with the sorted one
   setFlights(newFlights);

}, [flights])
```